src/storage/buffer/buffer_manager.cpp

```diff
@@ -42,11 +42,10 @@ BufferObj *BufferManager::Allocate(UniquePtr<FileWorker> file_worker) {
     auto buffer_obj = MakeUnique<BufferObj>(this, true, std::move(file_worker));
 
     auto res = buffer_obj.get();
-    std::unique_lock<std::shared_mutex> w_locker(rw_locker_);
+    std::unique_lock w_locker(rw_locker_);
     if (auto iter = buffer_map_.find(file_path); iter != buffer_map_.end()) {
-        UniquePtr<String> err_msg = MakeUnique<String>(fmt::format("BufferManager::Allocate: file {} already exists.", file_path.c_str()));
-        LOG_ERROR(*err_msg);
-        UnrecoverableError(*err_msg);
+        iter->second->Cleanup();
```

Comment:
Should raise exception.

Reply:
The file path is unique in all current files, but may duplicate with old files.
When cleanup task PickCleanup a file and remove it from catalog, the cleanup task will release lock. The actual cleanup task may process in parallel with adding file. The latter may has the same name with cleanuped file and try to Allocate here. So Cleanup the BufferObj here directly.